cirq/experiments/random_quantum_circuit_generation.py

```diff
@@ -237,12 +241,16 @@ def _single_qubit_layer(
         previous_single_qubit_layer: Dict['cirq.GridQubit', 'cirq.Gate'],
         prng: 'np.random.RandomState',
 ) -> Dict['cirq.GridQubit', 'cirq.Gate']:
+    if len(single_qubit_gates) == 1:
```

Comment:
So `len(set(single_qubit_gates))` would be safer here, but it diminishes the speedup down from 15x to around 3x.

Reply:
I added to the documentation that the given gates should not repeat.

Reply:
Another way would be two have two versions of `_single_qubit_layer`: one for the case with only one type of allowed single-qubit gate and one for the case with more than one type. You can then perform the check to detect which case is taking place before the main for loop of `random_rotations_between_grid_interaction_layers_circuit`. This might make it a small bit faster, too.

Reply:
Do you mean to have the two versions of `single_qubit_layer` have the same function call signature, even though the one with one type of allowed gate would not use any of its arguments? I think that would be too ugly. Any other way of doing it would require checking for which case we are in within the main loop anyway.

Reply:
I don't think that's too ugly at all. You need an internal API to accommodate the two cases, so there is nothing wrong with ignoring parameters not relevant in one of the cases. If you place the functions near each other it should be fairly clear to the reader why the unused arguments exist. Also, the practice is in fact sufficiently common that pylint has a specific mechanism to accommodate it (by `del`eting the argument) and google python style guide explicitly discusses this, see [1].

On the other hand, the current code has a severe failure mode that's going to be tricky to debug. Specifically, when one passes `[cirq.X, cirq.X]` for `single_qubit_gates` the code enters an infinite loop!

---
[1] http://go/pystyle#lint

Reply:
Very well, I have made the suggested change.